llvm/lib/Target/AIE/AIECombinerHelper.cpp

```diff
@@ -955,14 +957,35 @@ bool llvm::applySplatVector(MachineInstr &MI, MachineRegisterInfo &MRI,
   B.setInstrAndDebugLoc(MI);
   auto [DstVecReg, SrcReg] = MatchInfo;
   const LLT SrcTy = MRI.getType(SrcReg);
+  const LLT DstVecTy = MRI.getType(DstVecReg);
+  const unsigned DstVecSize = DstVecTy.getSizeInBits();
 
   if (SrcTy == LLT::scalar(8) || SrcTy == LLT::scalar(16)) {
     const LLT S32 = LLT::scalar(32);
     Register Src32BitReg = MRI.createGenericVirtualRegister(S32);
     B.buildAnyExt(Src32BitReg, SrcReg);
```

Comment:
CHECK: we have a guarantee that the element size matches the source size. i.e. the extension bits are really don't care.